src/coreclr/src/vm/threads.cpp

```diff
@@ -1539,6 +1539,8 @@ Thread::Thread()
     m_DeserializationTracker = NULL;
 
     m_currentPrepareCodeConfig = nullptr;
+
+    memset(dangerousObjRefs, 0, sizeof(dangerousObjRefs));
```

Comment:
Does this need to be under debug ifdef?

Reply:
Right, it does. I have not noticed that it is a debug build only thing.